android/libraries/rib-base/src/test/java/com/badoo/ribs/android/requestcode/RequestCodeBasedEventStreamImplTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.badoo.ribs.android.requestcode
+
+import com.badoo.ribs.core.Identifiable
+import com.badoo.ribs.plugins.RibsPlugins
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.verify
+import io.reactivex.disposables.CompositeDisposable
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+class RequestCodeBasedEventStreamImplTest {
```

Comment:
Add instrumentation tests to cover that it works as expected after rotation? That was the most common problem we tried to address.

Reply:
I've tried to emulate activity restart in the current tests but had no success with this. Android in a real environment never destroys the previous activities for the same app (it may happen only if we open another app). In espresso, we can restart the current activity but here we need to restart activity that is already in the back stack.

As we discussed earlier (when we faced emulator problems) it is necessary to develop tools that will supervise test running and probably it's not worth to do this and it will be enough to have only unit tests for this.

Reply:
Then maybe manually call save / detach / restore cycle on a test rib, set request code based request before cycle, check result is delivered after cycle? Every time it failed, it was related to above scenario. I wouldn't feel safe without covering it with tests somehow.

Reply:
Found a way how to ignore activity start in the test. A bit hacky but simple: I just overrode `startActivity` method, so it doesn't launch actual activity and then I manually call `onActivityResult`. Please check

Reply:
I've added one more test case that fails when we do not pass bundle to request code registry.
We are saving request code map to bundle only to avoid problems when we have a collision (in other cases it will work fine because we will generate the same group ids for the same RIB ids). I wrote a test that starts activity using ids that have different String values but have the same hash code.